.ado/scripts/prepublish-check.mjs

```diff
@@ -0,0 +1,86 @@
+// @ts-check
+import { spawnSync } from "node:child_process";
+import * as fs from "node:fs";
+import * as util from "node:util";
+
+/** @typedef {typeof import("../../nx.json")} NxConfig */
+
+/**
+ * @returns {NxConfig}
+ */
+function loadNxConfig(configFile = "nx.json") {
+  const nx = fs.readFileSync(configFile, { encoding: "utf-8" });
+  return JSON.parse(nx);
+}
+
+/**
+ * @returns {string}
+ */
+function getCurrentBranch() {
+  const { stdout } = spawnSync("git", ["rev-parse", "--abbrev-ref", "HEAD"]);
+  return stdout.toString().trim();
+}
+
+/**
+ * @param {string} branch
+ */
+function isStableBranch(branch) {
+  return /^\d+\.\d+-stable$/.test(branch);
+}
+
+/**
+ * @param {NxConfig["release"]} config
+ * @param {string} tag
+ * @param {string} [prerelease]
+ * @returns {asserts config is NxConfig["release"]}
+ */
+function enablePublishing(config, tag, prerelease) {
+  /** @type {string[]} */
+  const errors = [];
+
+  const { currentVersionResolverMetadata, preid } = config.version.generatorOptions;
+  if (preid !== prerelease) {
+    errors.push(`'release.version.generatorOptions.preid' must be set to '${prerelease || ""}'`);
+  }
+
+  if (currentVersionResolverMetadata.tag !== tag) {
+    errors.push(`'release.version.generatorOptions.currentVersionResolverMetadata.tag' must be set to '${tag}'`);
+  }
+
+  if (errors.length > 0) {
+    for (const e of errors) {
+      console.error("❌", e);
+    }
+    throw new Error("Nx Release is not correctly configured for the current branch");
+  }
+
+  console.log(`##vso[task.setvariable variable=publish_react_native_macos]1`);
+}
+
+function main({ "release-candidate": releaseCandidate }) {
+  const branch = getCurrentBranch();
+  if (!branch) {
+    throw new Error("Could not get current branch");
+  }
+
+  const { defaultBase, release } = loadNxConfig();
+  if (branch === defaultBase) {
+    enablePublishing(release, "nightly", "nightly");
+  } else if (isStableBranch(branch)) {
+    enablePublishing(release, "v" + branch, releaseCandidate ? "rc" : undefined);
+  }
+}
```

Comment:
I guess we don't have to change the base per stable branch, nice

Reply:
We still need to, but this script should fail if it's not correctly set.